src/main/java/com/exasol/projectkeeper/pom/MavenArtifactModelReader.java

```diff
@@ -1,22 +1,53 @@
 package com.exasol.projectkeeper.pom;
 
+import java.util.List;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.execution.MavenSession;
 import org.apache.maven.model.Model;
-import org.apache.maven.project.ProjectBuildingException;
+import org.apache.maven.project.*;
+import org.apache.maven.repository.RepositorySystem;
 
 /**
- * Interface for readers that read a maven {@link Model} from an artifact of the maven repository.
+ * Read a maven {@link Model} from an artifact of the maven repository using the {@link ProjectBuilder} that is injected
+ * by Maven core to Mojo.
  */
-public interface MavenArtifactModelReader {
+public class MavenArtifactModelReader {
+    private final ProjectBuilder mavenProjectBuilder;
+    private final MavenSession session;
+    private final RepositorySystem repositorySystem;
+
+    /**
+     * Create a new instance of {@link MavenArtifactModelReader}.
+     * 
+     * @param mavenProjectBuilder maven project builder
+     * @param session             maven session
+     * @param repositorySystem    maven repository system
+     */
+    public MavenArtifactModelReader(final ProjectBuilder mavenProjectBuilder, final MavenSession session,
```

Comment:
What is the artifact model? 
"An artifact is a file, usually a JAR, that gets deployed to a Maven repository."

Is the model a description of a specified JAR file?